### ISSUE TITLE
Tasks hang forever when run thorugh daemon in 3.5.X versions

### ISSUE THREAD

**Issue by malomehi** [Flexget/Flexget#3601]
### Expected behaviour:

Tasks can be executed through daemon the same way they are executed without it

### Actual behaviour:

Tasks hang forever when run thorugh daemon in 3.5.X versions

### Steps to reproduce:
- Start daemon with `flexget --loglevel verbose daemon start -d`
- Execute any task, even a dummy one with `flexget execute --tasks <task_name>`

The task execution starts but it does not progress. No new logs are displayed once the execution is started, even with `debug` or `trace` log level. No errors are logged, the task just keeps like waiting forever and never progress in the execution.
The tasks are perfectly executed if no daemon is running. The issue is only observed when the task is sent to a running daemon.
This issue is only observed in versions `3.5.0` and `3.5.1`. Going back to `3.4.2`resolves the issue.
No changes in the configuration have been made before updating to `3.5.0` or `3.5.1`. The same configuration works ok with `3.4.2`.

### Additional information:

- FlexGet version: `3.5.0` and `3.5.1`. Previous versions work as expected.
- Python version: `3.10.8`
- Installation method: `pip` and virtualenv
- Using daemon (yes/no): yes (all good without using daemon)
- OS and version: `Debian GNU/Linux bookworm/sid`. Kernel `Linux 6.0.0-2-amd64`
- Link to crash log:

**Comment by malomehi** [comment]
In order to provide further details, this is what happens when a task is executed without daemon (normal execution completes):
```
(flexget-daemon-test) odroid@debian: ~ $ flexget --loglevel debug execute --tasks cleanup
2022-11-04 14:23:33 DEBUG    manager                       Figuring out config load paths
2022-11-04 14:23:33 DEBUG    manager                       Found config: /home/odroid/.flexget/config.yml
2022-11-04 14:23:33 DEBUG    manager                       Config file /home/odroid/.flexget/config.yml selected
2022-11-04 14:23:33 DEBUG    manager                       sys.defaultencoding: utf-8
2022-11-04 14:23:33 DEBUG    manager                       sys.getfilesystemencoding: utf-8
2022-11-04 14:23:33 DEBUG    manager                       flexget detected io encoding: utf-8
2022-11-04 14:23:33 DEBUG    manager                       os.path.supports_unicode_filenames: False
2022-11-04 14:23:33 DEBUG    plugin                        Trying to load plugins from: ['/home/odroid/.flexget/plugins', '/home/odroid/.pyenv/versions/flexget-daemon-test/lib/python3.10/site-packages/flexget/plugins']
2022-11-04 14:23:35 DEBUG    plugin                        Plugin `memusage` requires plugin `ext lib `guppy`` to load.
2022-11-04 14:23:36 DEBUG    plugin                        Trying to load components from: ['/home/odroid/.flexget/components', '/home/odroid/.pyenv/versions/flexget-daemon-test/lib/python3.10/site-packages/flexget/components']
2022-11-04 14:23:38 DEBUG    plugin                        Plugins took 5.26 seconds to load. 322 plugins in registry.
2022-11-04 14:23:38 DEBUG    manager                       Connecting to: sqlite:////home/odroid/.flexget/db-config.sqlite
2022-11-04 14:23:38 DEBUG    manager                       config_name: config
2022-11-04 14:23:38 DEBUG    manager                       config_base: /home/odroid/.flexget
2022-11-04 14:23:38 DEBUG    variables                     trying to load variables from file
2022-11-04 14:23:38 DEBUG    variables                     updating DB with variable file contents
2022-11-04 14:23:39 DEBUG    manager                       New config data loaded.
2022-11-04 14:23:39 DEBUG    schema                        current flexget version already exist in db 3.5.1
2022-11-04 14:23:39 DEBUG    parsing                       setting default movie parser to internal. (options: {'guessit': <flexget.components.parsing.parsers.parser_guessit.ParserGuessit object at 0x7f62ff8a5f90>, 'internal': <flexget.components.parsing.parsers.parser_internal.ParserInternal object at 0x7f62ff8a6110>})
2022-11-04 14:23:39 DEBUG    parsing                       setting default series parser to internal. (options: {'guessit': <flexget.components.parsing.parsers.parser_guessit.ParserGuessit object at 0x7f62ff8a5f90>, 'internal': <flexget.components.parsing.parsers.parser_internal.ParserInternal object at 0x7f62ff8a6110>})
2022-11-04 14:23:39 DEBUG    cron_env                      Encoding utf-8 stored
2022-11-04 14:23:39 DEBUG    util.simple_persistence                 setting key terminal_encoding value 'utf-8'
2022-11-04 14:23:39 DEBUG    task_queue                    task queue shutdown requested
2022-11-04 14:23:39 VERBOSE  task_queue                    There are 1 tasks to execute. Shutdown will commence when they have completed.
2022-11-04 14:23:39 INFO     ipc.rpyc                      server started on [127.0.0.1]:35225
2022-11-04 14:23:39 DEBUG    task          cleanup         executing cleanup
2022-11-04 14:23:39 DEBUG    disable       cleanup         Disabled plugins: seen, seen_info_hash
2022-11-04 14:23:39 DEBUG    transmission  cleanup         Connecting to http://localhost:9091/transmission/
2022-11-04 14:23:39 DEBUG    backlog       cleanup         0 entries purged from backlog
2022-11-04 14:23:39 WARNING  details       cleanup         Task didn't produce any entries. This is likely due to a mis-configured or non-functional input.
2022-11-04 14:23:39 DEBUG    urlrewriter   cleanup         Checking 0 entries
2022-11-04 14:23:39 VERBOSE  details       cleanup         Summary - Accepted: 0 (Rejected: 0 Undecided: 0 Failed: 0)
2022-11-04 14:23:39 DEBUG    util.simple_persistence cleanup         Flushing simple persistence for task cleanup to db.
2022-11-04 14:23:40 DEBUG    task_queue                    task queue shut down
2022-11-04 14:23:40 INFO     ipc.rpyc                      listener closed
2022-11-04 14:23:40 DEBUG    util.simple_persistence                 Flushing simple persistence for task None to db.
2022-11-04 14:23:40 INFO     ipc.rpyc                      server has terminated
2022-11-04 14:23:40 DEBUG    manager                       Shutting down
2022-11-04 14:23:40 DEBUG    manager                       Removed /home/odroid/.flexget/.config-lock
(flexget-daemon-test) odroid@debian: ~ $
```
Then we start the daemon and run the same task:
```
(flexget-daemon-test) odroid@debian: ~ $ flexget --loglevel verbose daemon start -d
2022-11-04 14:25:19 INFO     manager                       Daemonizing...
2022-11-04 14:25:19 INFO     manager                       Daemonize complete. New PID: 103731
(flexget-daemon-test) odroid@debian: ~ $ flexget --loglevel debug execute --tasks cleanup
2022-11-04 14:25:31 DEBUG    manager                       Figuring out config load paths
2022-11-04 14:25:31 DEBUG    manager                       Found config: /home/odroid/.flexget/config.yml
2022-11-04 14:25:31 DEBUG    manager                       Config file /home/odroid/.flexget/config.yml selected
2022-11-04 14:25:31 DEBUG    manager                       sys.defaultencoding: utf-8
2022-11-04 14:25:31 DEBUG    manager                       sys.getfilesystemencoding: utf-8
2022-11-04 14:25:31 DEBUG    manager                       flexget detected io encoding: utf-8
2022-11-04 14:25:31 DEBUG    manager                       os.path.supports_unicode_filenames: False
There is a FlexGet process already running for this config, sending execution there.
2022-11-04 14:25:31 DEBUG    manager                       Sending command to running FlexGet process: ['--loglevel', 'debug', 'execute', '--tasks', 'cleanup']
2022-11-04 14:25:32 DEBUG    cron_env                      Encoding utf-8 stored
2022-11-04 14:25:32 DEBUG    util.simple_persistence                 setting key terminal_encoding value 'utf-8'
```
The tasks keeps executing forever but it never progresses. No errors are displayed. And no logs. It just hangs.

**Comment by gazpachoking** [comment]
It looks like this is to do with something that changed between RPyC version 5.1.0 and 5.2.1. I'll try to look into this, or revert our rpyc version later tonight.

**Comment by gazpachoking** [comment]
Didn't have a chance to look into the actual issue yet, but it should be fixed with the next release by lowering our version of rpyc. We really need to figure out a way to add this sort of stuff to the test suite so we can catch it earlier.

**Comment by gazpachoking** [comment]
My inclination is that this is a new bug in RPyC, but debugging is harder than expected. If I disable the log and stdout transport from daemon to client, everything works, but when I enable it things just stop running at some point, seemingly with no exceptions raised.

**Comment by comrumino** [comment]
This issue should be resolved in [RPyC 5.3.1](https://github.com/tomerfiliba-org/rpyc/releases/tag/5.3.1) that I released today. Of course, my testing was rather minimal.
## Test files
My flexget `config.yml`:
```
tasks:
  digest 1:
    mock:
      - title: entry 1
    accept_all: yes
    digest: aoeu
```
Shell script to reproduce the issue and report pass/fail.
```
#!/usr/bin/env bash
function ansi_wrap_msg() { printf "\e[1;${1}m${2}\e[m"; }
function blue_msg() { ansi_wrap_msg "34" "$1"; }
function red_msg() { ansi_wrap_msg "31" "$1"; }
function getpid() {
     printf "%s" "$(ps -aux | grep 'versions/3[^/]*/bin/flexget' | grep -v vim | awk '{print $2}' | tr -d '\s')"
}
function testflex() {
    timeout 5 flexget execute --tasks 'digest 1' && blue_msg "exit code $?: task-passed\n"
}
pid=$(getpid)
[[ "$pid" =~ ^[0-9]+$ ]] && kill -9 "$pid"
printf "" > "/tmp/rpyc.log"
flexget --loglevel DEBUG daemon start -d
testflex || red_msg "exit code $?: task-failed\n"
pid=$(getpid)
[[ "$pid" =~ ^[0-9]+$ ]] && kill -9 $pid
```
My hooked entrypoint for flexget to enable RPyC logs
```
#!/home/myuser/.pyenv/versions/3.8-dev/bin/python3.8
# -*- coding: utf-8 -*-
import re
import sys

import logging
import rpyc
logger = rpyc.setup_logger(logfile='/tmp/rpyc.log')
rpyc.core.protocol.DEFAULT_CONFIG['logger'] = logger
rpyc.core.protocol.DEFAULT_CONFIG['bind_threads'] = True
from flexget import main

if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
    sys.exit(main())
```
The RPyC logs were not very insightful/helpful. Ultimately, I located the commit that introduced the breaking change. It came down to the ordering of lock release, dispatching msg/reply, and notify event that the lock was released. 

I'm haven't translated your use case into a minimal unit test to prevent regressions yet... I would appreciate any help on  creating a minimal test case that would not require flexget being installed.
